devlib/target.py

```diff
@@ -1018,19 +1015,19 @@ async def read_bool(self, path):
         return await self.read_value.asyn(path, kind=boolean)
 
     @asyn.asynccontextmanager
-    async def revertable_write_value(self, path, value, verify=True):
+    async def revertable_write_value(self, path, value, verify=True, as_root=True):
         orig_value = self.read_value(path)
         try:
-            await self.write_value.asyn(path, value, verify)
+            await self.write_value.asyn(path, value, verify, as_root)
             yield
         finally:
-            await self.write_value.asyn(path, orig_value, verify)
+            await self.write_value.asyn(path, orig_value, verify, as_root)
```

Comment:
At this point we might want to pass the parameters using named syntax:
```
self.write_value.asyn(path, orig_value=orig_value, verify=verify, as_root=as_root)
```
Otherwise we run a relatively high risk of mixing parameters, and this can easily backfire especially for booleans (e.g. passing the value of `orig_value` to `verify` is likely to work, as a non-empty string and non-zero int will be considered as `True` (`if  x` is equivalent to `if bool(x)`))